test/sink.test.js

```diff
@@ -42,3 +45,113 @@ test('.writer() - no value for "type" argument - should throw', () => {
         sink.writer();
     }).toThrowError('"type" is missing');
 });
+
+test.only('.writer() - happy path', async done => {
+    expect.assertions(2);
+    const sink = getValidSink();
+
+    await sink.ready;
+
+    const gcsFakeWritestream = new stream.Writable({
+        _data: false,
+        write(chunk, encoding, next) {
+            this._data += chunk;
+            next();
+        },
+    });
+
+    sink.gcs._setState({
+        createWriteStream: gcsFakeWritestream,
+        move: null,
+    });
+
+    const dest = sink.writer('json');
+
+    const source = require('fs').createReadStream(
+        `${__dirname}/mock/feed.a.json`
+    );
+
+    source.on('error', done);
+    dest.on('error', done);
+    gcsFakeWritestream.on('error', done);
+
+    source.pipe(dest);
+
+    dest.on('file not saved', () => {
+        done(new Error('File not saved'));
+    });
+
+    dest.on('file saved', (id, file) => {
+        expect(id).toMatchSnapshot();
+        expect(file).toMatchSnapshot();
+        process.nextTick(done);
+    });
+});
+
+test('.get() - should return null when ', async () => {
+    const sink = getValidSink();
+
+    await sink.ready;
+
+    const download = 'some-file-content';
+    sink.gcs._setState({
+        download,
+    });
+
+    const result = await sink.get('some-file');
+    expect(result).toBe(download);
+});
+
+test('.set() - should return true if success', async () => {
+    const sink = getValidSink();
+
+    await sink.ready;
+
+    sink.gcs._setState({ save: '' });
+
+    const result = await sink.set('some-file', 'file-content');
+
+    expect(result).toBe(true);
+});
+
+test('.set() - should return true if success', async () => {
+    const sink = getValidSink();
+
+    await sink.ready;
+
+    sink.gcs._setState({ save: '' });
+
+    const result = await sink.set('some-file', 'file-content');
+
+    expect(result).toBe(true);
+});
```

Comment:
This test is identical to the test above. This won't help you get more test coverage 😁

Reply:
delirious here...